pkgs/development/python-modules/poetry-core/default.nix

```diff
@@ -54,7 +54,9 @@ buildPythonPackage rec {
     pytest-mock
     pytestCheckHook
     setuptools
-    tomlkit
+    # tomlkit 0.11.6 is the last version to not rely on poetry to build,
+    # so we use it here to prevent a circular dependency
+    tomlkit_11_6
```

Comment:
That's not right, there's nothing related to tomlkit newly depending on poetry in the 0.11.7 release: https://github.com/sdispater/tomlkit/compare/0.11.6...0.11.7

The error you ran into must've been this one:
```
this derivation will be built:
  /nix/store/cgyhiwnccdb2lh4q3di996iv35yk4hdf-python3.10-tomlkit-0.11.7.drv
building '/nix/store/cgyhiwnccdb2lh4q3di996iv35yk4hdf-python3.10-tomlkit-0.11.7.drv'...
[...]
building
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "/build/tomlkit-0.11.7/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "/nix/store/jhflvwr40xbb0xr6jx4311icp9cym1fp-python3-3.10.12/lib/python3.10/tokenize.py", line 394, in open
    buffer = _builtin_open(filename, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
/nix/store/7aprjn0yf6p06krg4087sxvilfp4c8yy-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
```

For some reason there was a `setup.py` before but now there isn't anymore. I could reproduce this by just fetching the `tomlkit-0.11.6.tar.gz` from https://pypi.org/project/tomlkit/0.11.6/#files, and comparing it to the one from https://pypi.org/project/tomlkit/0.11.7/#files. The latter indeed doesn't have a `setup.py` file anymore.

I really don't know why this changed or why the `setup.py` was even generated originally, but this looks like some upstream inconsistency which shouldn't have happened.

Can you file an upstream bug for that? I think we should just downgrade to 0.11.6 until that's resolved then.

Reply:
> but this looks like some upstream inconsistency which shouldn't have happened.

Why do you say so? They're using `pyproject.toml` to manage their packaging, so there's no need for a `setup.py`. In fact, the PyPA's sample project only has `pyproject.toml` as of last year: https://github.com/pypa/sampleproject/commit/aeeb50a948addcb712ad4261df472263514991e1

They might not have announced it as a breaking change on 0.11.7 so, sure, it might be an inconsistency. But I don't think it's worth a bug report upstream unless I'm missing something.

Reply:
> The error you ran into must've been this one:

Not quite, the error you are referring to is the error you will get if you build `tomlkit` with `format` as `setuptools` instead of `pyproject` (which was also updated on this PR)

Reply:
Something changed so that no `setup.py` is generated anymore, and this change is not recorded in the git history nor documented anywhere. We don't really know what occurred there, and we shouldn't try to work around it with more code unless we know it's the right solution. Maybe `setup.py` was explicitly provided to make bootstrapping poetry easier?

Reply:
Filed https://github.com/sdispater/tomlkit/issues/305

Will revert this PR to the original PR with a comment linking to that issue.

Reply:
Let's wait a bit to see if the issue gets resolved (I see you found a good lead)

Reply:
@infinisil, would you be open to us landing this as is? Telegraf is currently pretty much unusable because of the `pkgs.format.toml` issue.

I'm also not super sure that setting `tool.poetry.build.generate-setup-file` upstream is a reasonable solution, since that option is going to be removed from `poetry` in the future (see https://python-poetry.org/blog/announcing-poetry-1.3.0/#generate-setup-file--false).

A solution might be for us to allow for Python packages with the `pyproject` `format` to be built with `pip` instead of `poetry` within `nixpkgs`. But I haven't looked into how that would happen yet.

Reply:
We have these options now:

1. Pin to 0.11.4 (current state of this PR)
2. Upgrade to 0.11.8 and apply a patch which starts generating `setup.py`
3. Upgrade to 0.11.8 and apply a patch which moves from using `poetry` to using `setuptools` (see my most recent comment on https://github.com/sdispater/tomlkit/pull/306)
4. Wait until a new release of `tomlkit` comes out with a fix

I'm personally leaning towards 2, since it's the least intrusive change to `nixpkgs` (and hence simpler to remove once we figure out the solution upstream).

What do you think?

Reply:
Oh yeah good point, I like number 2, but still I'd like to give upstream some time to respond. If you need the fix immediately you can still apply it as a patch to Nixpkgs, or use overlays.